portal/app/api/v1/endpoint/farm_data_parser.py

```diff
@@ -0,0 +1,24 @@
+import json
+import pandas as pd
+import datetime
+
+
+def farm_data_parser(argv):
+    """
+        Parses json blob and returns the argument fields as individual dictionaries
+        in json form
+    """
+    # sets serialized_farm_data to data
+    data = argv[-1]
+    # pulls device data from the json
+    farm_data = [ json.loads(row["device_data"]) for row in data ]
+    # creates dataframe with argument fields as columns
+    farm_df = pd.DataFrame(farm_data)[argv[:-1]]
+
+    try:
+        farm_df["timestamp"] = pd.to_datetime(farm_df["timestamp"])
+    except KeyError:
+        print("Timestamp column not found.")
+
+    # prepares json to be sent to javascript
+    return farm_df.to_json(date_format='iso')
```

Comment:
if this indeed throws an exception, will this line still succeed? If so, how and then, why catch the exception at all?

If not, then you need to handle that since all you are doing is printing the a message and then continuing execution.

Reply:
That was in there because of the confusion I had about trying to display the wrong data. I removed that and created an exception in views/base.py, incase the data coming isn't valid. Tested by querrying device_farm_id__lte = 18 instead of 16.